### ISSUE TITLE
Add generic object cache for Velocity

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#23037]
Customers are always looking for an easy way to pull content / data from a 3rd party system and save it for further use in  dotCMS.  This PR provides both a viewtool and a Block Directive that can be used to populate your caches.

### Viewtool cache
The viewtool cache is pretty straight forward - it can be used to set and get values that are stored in dotCMS memory for later retrieval. 
```Velocity
#set($testing = $dotcache.get("my-cache-key"))
#if(!$testing)
    #set($testing = $json.fetch("https://demo.dotcms.com/api/content/render/false/query/+contentType:Activity%20+deleted:false%20+working:true/orderby/modDate%20desc/limit/4"))
    ## cache for 5 minutes
    $dotcache.put("my-cache-key", $testing, 600)
    ## or cache forever
    $dotcache.put("my-cache-key", $testing)
#end
<ul>
    #foreach($con in $testing.contentlets)
       <li>$con.title</li> 
    #end
</ul>
```
If you need to remove it from cache:
```Velocity
## remove from cache
$dotcache.remove("my-cache-key")
```

### Block Directive Cache
In dotCMS, we already have a block cache that lets you wrap your output in a cache block so that it does not need to be re-rendered in velocity.  This is helpful, but there are times when you want to cache more than just the text output.  Many times, you want to cache an object in velocity for use by multiple requests - especially when pulling data from remote systems.  For example:

```Velocity
## cache for 5 minutes
#dotcache("my-new-cache-key",600)
    #set($testDate = $date)
    #set($testing = $json.fetch("https://demo.dotcms.com/api/content/render/false/query/+contentType:Activity%20+deleted:false%20+working:true/orderby/modDate%20desc/limit/4"))<br>

    #spit out some text that will be cached
    cached: $testDate<br>
#end
----
The above was cached at: $testDate<br>
It is now: $date
<ul>
    #foreach($con in $testing.contentlets)
       <li>$con.title</li> 
    #end
</ul>
```

will render the values from cache (one minute later):

```
cached: Nov 14, 2022, 9:02:11 PM
----
The above was cached at: Nov 14, 2022, 9:02:11 PM
It is now: Nov 14, 2022, 9:03:17 PM
- Snowboarding
- Hiking
- Zip-Lining
- Skiing
```

**Comment by rjvelazco** [comment]
**Passed Internal QA**

Docker Image: `23.01_a953eb1e_SNAPSHOT`

**Comment by bryanboza** [comment]
Fixed, tested on release-23.01 // Docker // FF
